libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -289,7 +289,7 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
 - (BOOL)shouldUpdateBaseUrl {
     // only set base URL to apiURL if baseURL is not already specified as absolute URL with it's own host
     // this check is necessary as there are salesforce URL that is content server based and not API based
-    return (!self.baseURL.scheme && !self.baseURL.host) || (self.baseURL == self.cachedAPIURL);
+    return (!self.baseURL.scheme && !self.baseURL.host) || [self.baseURL isEqual:self.cachedAPIURL];
```

Comment:
Both of these properties use "copy"...so there's a good chance "==" would not return true because they could be different instances.